hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/serialization/MapEntryHook.java

```diff
@@ -23,15 +23,16 @@
 import com.hazelcast.nio.serialization.StreamSerializer;
 
 import java.io.IOException;
+import java.util.AbstractMap;
 import java.util.Map.Entry;
 
 import static com.hazelcast.jet.Util.entry;
 
-public final class MapEntryHook implements SerializerHook<Entry> {
+public final class MapEntryHook implements SerializerHook<AbstractMap.SimpleImmutableEntry> {
```

Comment:
maybe we can add `AbstractMap.SimpleImmutableEntry` serializer to IMDG side so we don't maintain any serializers on our side ?

Reply:
this was a work-around, just to make builds pass till we find a real fix for the issue on imdg side. but even with this change there are some test failures
`Failed to serialize 'com.hazelcast.query.impl.QueryEntry'`